File: RepoTransBench/source_projects/JavaScript/creationix_step/public_tests/callback.public.test.js
require('./helper');

// Use a *different* file for content testing to change test data.
// We'll use the license.txt file (license.txt always present, distinct from __filename in existing test).

var licenseText = fs.readFileSync("license.txt", 'utf8');

// Testing Step passing async/sync results to next layer
expect('uno');
expect('dos');
expect('tres');
Step(
  function readLicense() {
    fulfill("uno");
    fs.readFile("license.txt", 'utf8', this);
  },
  function capitalize(err, text) {
    fulfill("dos");
    if (err) throw err;
    assert.equal(licenseText, text, "License Text Loaded");
    return text.split('').reverse().join('');
  },
  function showIt(err, newText) {
    fulfill("tres");
    if (err) throw err;
    assert.equal(licenseText.split('').reverse().join(''), newText, "License Text Reversed");
  }
);

File: RepoTransBench/target_projects/JavaScript/Python/creationix_step/public_tests/callback_public_test.py
import os
from public_tests.helper import expect, fulfill, assert_ as assert_builtin

def test_callback_public():
    license_file = os.path.abspath("license.txt")
    with open(license_file, 'r', encoding='utf-8') as f:
        license_text = f.read()

    expect('uno')
    expect('dos')
    expect('tres')

    def read_license(cb):
        fulfill("uno")
        with open(license_file, 'r', encoding='utf-8') as lf:
            cb(None, lf.read())
    def capitalize(err, text, cb):
        fulfill("dos")
        if err:
            raise err
        assert_builtin.equal(license_text, text, "License Text Loaded")
        return text[::-1]
    def show_it(err, new_text, cb):
        fulfill("tres")
        if err:
            raise err
        assert_builtin.equal(license_text[::-1], new_text, "License Text Reversed")

    def Step(*funcs):
        queue = list(funcs)
        def callback(*args):
            if not queue:
                if args and args[0]:
                    raise args[0]
                return
            func = queue.pop(0)
            try:
                if len(args) == 0:
                    result = func(callback)
                else:
                    result = func(*args, callback)
                if result is not None:
                    callback(None, result)
            except Exception as e:
                callback(e)
        callback()

    Step(read_license, capitalize, show_it)